### PR TITLE
Migrate TrailingWhitespace from AstNode.

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/TrailingWhitespaceCheck.java [hunk 1]
```diff
@@ -57,19 +58,30 @@ public void setCharset(Charset charset) {
...
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public void visitFile(Tree scriptTree) {
     List<String> lines;
+
     try {
       lines = Files.readLines(getContext().getFile(), charset);
```
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/TrailingWhitespaceCheck.java]: @lindamartin I would include only that line in the try block.

Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/TrailingWhitespaceCheck.java]: @pynicolas why so ?

Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/TrailingWhitespaceCheck.java]: @lindamartin I generally try to make a try block rather small: I prefer to make it obvious where the exception can come from.

Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/TrailingWhitespaceCheck.java]: @pynicolas agreed, updated in 457cb22a72d58b734e08e93bda4429d7bcf667e3.

**Comment by lindamartin** [comment]
Merged in dcc3c48ba456b760d90673d8d28deca62c7b620e.

**Pull request closed without merging**